src/Compilers/CSharp/Test/Symbol/Compilation/SemanticModelAPITests.cs

```diff
@@ -2101,6 +2101,18 @@ void M(int x)
             VerifySpeculativeSemanticModelForMethodBody(blockStatement, speculativeModel);
         }
 
+        [Fact()]
+        [WorkItem(10211, "https://github.com/dotnet/roslyn/issues/10211")]
+        public void GetDependenceChainRegression_10211()
+        {
+            var compilation = CreateCompilation(@"
+class Child : Parent {}
+class Parent {}
+");
+            var semanticModel = compilation.GetSemanticModel(compilation.SyntaxTrees[0]);
+            semanticModel.GetMethodBodyDiagnostics();
```

Comment:
There are no diagnostics that should be reported.  `GetMethodBodyDiagnostics` simply shouldn't throw an exception.

I've added a comment: `// Ensure that we don't crash here.`